tests/unit/php/Core/Checkout/Customer/SalesChannel/RegisterConfirmRouteTest.php

```diff
@@ -0,0 +1,211 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Tests\Unit\Core\Checkout\Customer\SalesChannel;
+
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\MockObject\Stub;
+use PHPUnit\Framework\TestCase;
+use Shopware\Core\Checkout\Customer\CustomerCollection;
+use Shopware\Core\Checkout\Customer\CustomerEntity;
+use Shopware\Core\Checkout\Customer\Exception\CustomerAlreadyConfirmedException;
+use Shopware\Core\Checkout\Customer\SalesChannel\CustomerResponse;
+use Shopware\Core\Checkout\Customer\SalesChannel\RegisterConfirmRoute;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepositoryInterface;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\EntitySearchResult;
+use Shopware\Core\Framework\Validation\DataBag\RequestDataBag;
+use Shopware\Core\Framework\Validation\DataValidationDefinition;
+use Shopware\Core\Framework\Validation\DataValidator;
+use Shopware\Core\Framework\Validation\Exception\ConstraintViolationException;
+use Shopware\Core\System\SalesChannel\Context\SalesChannelContextPersister;
+use Shopware\Core\System\SalesChannel\Context\SalesChannelContextServiceInterface;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Symfony\Component\Validator\Constraints\IsTrue;
+use Symfony\Component\Validator\ConstraintViolationList;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+
+/**
+ * @internal
+ * @group rules
+ * @covers \Shopware\Core\Checkout\Customer\SalesChannel\RegisterConfirmRoute
+ */
+class RegisterConfirmRouteTest extends TestCase
+{
+    /**
+     * @var MockObject|SalesChannelContext
+     */
+    protected $context;
+
+    /**
+     * @var MockObject|EventDispatcherInterface
+     */
+    protected $eventDispatcher;
+
+    /**
+     * @var MockObject|EntityRepositoryInterface
+     */
+    protected $customerRepository;
+
+    /**
+     * @var MockObject|DataValidator
+     */
+    protected $validator;
+
+    /**
+     * @var Stub|SalesChannelContextServiceInterface
+     */
+    protected $salesChannelContextService;
+
+    /**
+     * @var RegisterConfirmRoute
+     */
+    protected $route;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->context = $this->createMock(SalesChannelContext::class);
+        $this->eventDispatcher = $this->createMock(EventDispatcherInterface::class);
+        $this->customerRepository = $this->createMock(EntityRepositoryInterface::class);
+        $this->validator = $this->createMock(DataValidator::class);
+        $this->salesChannelContextPersister = $this->createMock(SalesChannelContextPersister::class);
```

Comment:
One last change request: This property is still undefined. Please fix this so we can schedule your change to be merged.